tests/Integration/Behaviour/Features/Scenario/Order/order_shipping_recalculate.feature

```diff
@@ -54,17 +55,81 @@ Feature: Order from Back Office (BO)
+      | total_shipping_tax_excl  | 2.00   |
+      | total_shipping_tax_incl  | 2.12   |
+    Then order "bo_order1" has 0 payments
 
-  Scenario: With PS_ORDER_RECALCULATE_SHIPPING = 0, check the total price is not recalculated
+  Scenario: With PS_ORDER_RECALCULATE_SHIPPING = 1 & status = Processing in progress, check the total price is recalculated
+    Given shop configuration for "PS_ORDER_RECALCULATE_SHIPPING" is set to 1
+    And there is a product in the catalog named "Shipping Product" with a price of 15.0 and 100 items in stock
+    And product "Shipping Product" weight is 0.63 kg
+    And I add 1 products "Shipping Product" to the cart "dummy_cart"
+    And I select carrier "weight_carrier" for cart "dummy_cart"
+    Then cart "dummy_cart" should have "weight_carrier" as a carrier
+    When I add order "bo_order1" with the following details:
+      | cart                | dummy_cart                 |
+      | message             | test                       |
+      | payment module name | dummy_payment              |
+      | status              | Processing in progress     |
+    Then order "bo_order1" should have 3 products in total
+    And order "bo_order1" should have 1 invoice
+    And order "bo_order1" should have 0 cart rule
+    # Carrier less expensive is chosen by default
+    And order "bo_order1" should have "weight_carrier" as a carrier
+    And order "bo_order1" should have following details:
+      | total_products           | 38.800 |
+      | total_products_wt        | 41.130 |
+      | total_discounts_tax_excl | 0.0    |
+      | total_discounts_tax_incl | 0.0    |
+      | total_paid_tax_excl      | 43.800 |
+      | total_paid_tax_incl      | 46.430 |
+      | total_paid               | 46.430 |
+      | total_paid_real          | 46.430 |
+      | total_shipping_tax_excl  | 5.0    |
+      | total_shipping_tax_incl  | 5.30   |
+    When I update order "bo_order1" Tracking number to "TEST1234" and Carrier to "default_carrier"
+    Then order "bo_order1" should have "default_carrier" as a carrier
+    ## Shipping equals to 0 as 2 products in cart have no weight
+    And order "bo_order1" should have following details:
+      | total_products           | 38.800 |
+      | total_products_wt        | 41.130 |
+      | total_discounts_tax_excl | 0.0    |
+      | total_discounts_tax_incl | 0.0    |
+      | total_paid_tax_excl      | 45.800 |
+      | total_paid_tax_incl      | 48.550 |
+      | total_paid               | 48.550 |
+      | total_paid_real          | 46.430 |
+      | total_shipping_tax_excl  | 7.0    |
+      | total_shipping_tax_incl  | 7.42   |
+    Then order "bo_order1" has 1 payments
+    And order "bo_order1" payment in first position should have the following details:
+      | paymentMethod  | Dummy payment   |
+      | transactionId  |                 |
+      | amount         | $46.43          |
+      | invoiceNumber  | #IN000001       |
+    ## Change to paid order status
```

Comment:
:warning: I don't change this behavior. 

It's done in `classes/order/OrderHistory` (https://github.com/PrestaShop/PrestaShop/blob/develop/classes/order/OrderHistory.php#L373-L410),
* When you change the status of an order, 
* When the new order status is "Paid"
* When there are already invoices
* When there is a "rest to be paid",
* Then a new order payment is added 

This behavior exists since (nearly) 10 years (1.5.5.0) => https://github.com/PrestaShop/PrestaShop/commit/ccb07311262bf06b9b06180a7f10f9dd9d82cc69#diff-32c0a9dbcd2665ee1a3e6258057b294d4653a6f84a4453cc32c4886ee2e9f2d7.

Reply:
Ok, my bad ^^